vclib/src/commonMain/kotlin/at/asitplus/wallet/lib/agent/Holder.kt

```diff
@@ -87,48 +90,61 @@ interface Holder {
      * has been set with [setRevocationList]
      */
-    suspend fun getCredentials(
-        credentialSchemes: Collection<ConstantIndex.CredentialScheme>? = null,
-    ): Collection<StoredCredential>?
-
-    sealed class StoredCredential {
-        data class Vc(
-            val vcSerialized: String,
-            val vc: VerifiableCredentialJws,
-            val status: Validator.RevocationStatus
-        ) : StoredCredential()
-
-        data class SdJwt(
-            val vcSerialized: String,
-            val sdJwt: VerifiableCredentialSdJwt,
-            val status: Validator.RevocationStatus
-        ) : StoredCredential()
-
-        data class Iso(
-            val issuerSigned: IssuerSigned
-        ) : StoredCredential()
+    suspend fun getCredentials(): Collection<StoredCredential>?
+
+    sealed class StoredCredential(
+        open val storeEntry: SubjectCredentialStore.StoreEntry,
+        val status: Validator.RevocationStatus,
+    ) {
+        class Vc(
+            override val storeEntry: SubjectCredentialStore.StoreEntry.Vc,
+            status: Validator.RevocationStatus
+        ) : StoredCredential(
+            storeEntry = storeEntry,
+            status = status
+        )
+
+        class SdJwt(
+            override val storeEntry: SubjectCredentialStore.StoreEntry.SdJwt,
+            status: Validator.RevocationStatus
+        ) : StoredCredential(
+            storeEntry = storeEntry,
+            status = status
+        )
+
+        class Iso(
+            override val storeEntry: SubjectCredentialStore.StoreEntry.Iso,
+            status: Validator.RevocationStatus
+        ) : StoredCredential(
+            storeEntry = storeEntry,
+            status = status
+        )
     }
 
+    data class HolderResponseParameters(
+        val presentationSubmission: PresentationSubmission,
+        val verifiablePresentations: List<CreatePresentationResult>
+    )
+
     /**
-     * Creates a [VerifiablePresentation] serialized as a JWT for all the credentials we have stored,
-     * that match the [credentialSchemes] (if specified). Optionally filters by [requestedClaims] (e.g. in ISO case).
-     *
-     * May return null if no valid credentials (i.e. non-revoked, matching attribute name) are available.
+     * Creates an array of [VerifiablePresentation] and a [PresentationSubmission] to match
+     * the [presentationDefinition].
      */
     suspend fun createPresentation(
         challenge: String,
         audienceId: String,
-        credentialSchemes: Collection<ConstantIndex.CredentialScheme>? = null,
-        requestedClaims: Collection<String>? = null,
-    ): CreatePresentationResult?
+        presentationDefinition: PresentationDefinition,
+        // TODO: add authorization semantics to detect unauthorized requests
+        // - eg. a service provider asking for an attribute he should not be allowed to see
```

Comment:
imho that's not part of the logic in vclib, especially not at the holder interface

Reply:
In my opinion it would be actor performing the selective disclosure procedure that must be the one who is aware of the verifiers permissions in accessing them..
Alternatively there would need to be a separate check before or after the selective disclosure has been performed, and in this case the TODO would have to be moved somewhere else?

Reply:
You're right, there should be something like a verification method (or function pointer) in the holder class, that Wallets need to implement (with the default implementation to allow all requests).

Reply:
Todo for the future

Reply:
see https://github.com/a-sit-plus/kmm-vc-library/issues/60